snippets/survival_analysis/Coxph_model/Coxph_survival_regression.Rmd

```diff
@@ -8,61 +8,90 @@ author: "Jana Fischer"
 knitr::opts_chunk$set(echo = TRUE)
 ```
 
-# Read in data
+# Read in survival data
 ```{r}
 library(data.table)
+library(survival)
+library(ggplot2)
 
-surv_dat = fread('/home/jana/Github/bbsnippets/data/survival_example_data.csv',header = T)
+surv_dat = fread('~/Github/bbsnippets/data/survival_example_data.csv',header = T)
 head(surv_dat)
 
-# Column description:
-## PID: patient identifier
-## grade: tumor pathology grade
-## OSmonth: overall suvival from time of diagnosis [months]
-## DFSmonth: disease free survival from time of diagnosis [months]
-## Patientstatus: life/death status 
-### "death by primary disease" = death due to disease
-### "death" = disease unrelated death
-### "alive" = alive at given survival time
-### "alive w metastases" = alive with metasatses at given survival time
-
 ```
 
+Column description:  
+ PID: patient identifier  
+ grade: tumor pathology grade  
+ OSmonth: overall suvival from time of diagnosis [months]  
+ DFSmonth: disease free survival from time of diagnosis [months]  
+ Patientstatus: life/death status   
+  "death by primary disease" = death due to disease  
+  "death" = disease unrelated death  
+  "alive" = alive at given survival time  
+  "alive w metastases" = alive with metasatses at given survival time  
+
+
 # Prepare censoring information
-```{r}
 
-#Censoring for overall survival (1 = event (death due to disease), 0 = censored)
+Survival data is censored when the event has not yet occurred for a patient
+at a given time or the patient is lost to follow-up during the study.
+A more detailed explanation can be found here: 
+http://www.sthda.com/english/wiki/survival-analysis-basics#censoring
+
+```{r}
+# Censoring for overall survival (1 = event (death due to disease), 0 = censored)
 surv_dat$censoringOS = 0
 surv_dat$censoringOS[surv_dat$Patientstatus == 'death by primary disease'] = 1
 
-#Censoring for disease free survival (1 = event (recurrence), 0 = censored)
+# Censoring for disease free survival (1 = event (recurrence), 0 = censored)
 surv_dat$censoringDFS = 0
 surv_dat$censoringDFS[surv_dat$Patientstatus == 'alive w metastases'] = 1
 surv_dat$censoringDFS[surv_dat$OSmonth > surv_dat$DFSmonth] = 1 
 
 ```
 
-# Coxph survival model for grade
-```{r}
-library(survival)
-library(ggplot2)
+# Coxph survival model for grade groups
+
+Covariates passed to the coxph model can be continuous or categorical.
+With categorical covariates each level is considered a distinct group
+and the first level is used as reference.
 
+```{r}
 #Set reference level for grade groups (first level)
 surv_dat$grade = factor(surv_dat$grade,levels = 1:3)
 
-#Calculate coxph model
+#Calculate coxph model. The formula requires a standard survival object
+#built with the Surv function from the "survival" package. In the survival
+#object a plus behind the time indicates censoring.
 res = coxph(Surv(surv_dat$OSmonth, surv_dat$censoringOS) ~ grade , data =  surv_dat)
```

Comment:
Cant this be simplified to:
```
res = coxph(Surv(OSmonth, censoringOS) ~ grade , data =  surv_dat)
```

Reply:
Nope, unfortunately that doesn't work. The Surv function seems not to take columns from the data that is passed to coxph.

Reply:
Okay, I didnt test it but saw that they show it to work like this in the coxph documentation examples: https://www.rdocumentation.org/packages/survival/versions/3.1-11/topics/coxph

Reply:
hmm they use a list... maybe it only doesn't work with data.tables?

Reply:
Never mind, it does work! Just had a typo.